net/src/parse.rs

```diff
@@ -59,6 +59,10 @@ pub(crate) trait ParsePayloadWith {
     fn parse_payload_with(&self, param: &Self::Param, cursor: &mut Reader) -> Option<Self::Next>;
 }
 
+#[derive(thiserror::Error, Debug)]
+#[error("Maximum legal packet buffer size is 2^16 (requested {0})")]
+pub struct IllegalBufferLength(pub usize);
```

Comment:
THere is no std::io or some Rust way of saying buffer to larger or something like that?

Reply:
I could likely recycle something from std::io.  That would give less control over the generated error message.

I kinda expect we will never see this error (but I won't say that too loud tho because the programming gods will strike me down)